### ISSUE TITLE
Issue Launching with Chocolatey Licensed

### ISSUE THREAD

**Issue by TheVakman** [chocolatey/ChocolateyGUI#469]
![chocolateyguierror](https://user-images.githubusercontent.com/18545156/30644391-34e5ba46-9de0-11e7-9783-073a980f1c6b.PNG)

The attached screenshot is the error I get on the first install of 0.14.300. This occurs whether it is launched as an administrator or a non-admin user. 
This error does not occur on the 0.14.290 release. Perhaps, this is a problem with the licensed extension?

On subsequent launches, it closes instantly and in the log:

```
2017-09-20 08:43:57.232 -04:00 [Fatal] Packages failed to load
2017-09-20 08:43:57.234 -04:00 [Fatal] Local source control view model failed to load.
2017-09-20 08:43:57.235 -04:00 [Fatal] Unhandled Exception
2017-09-20 08:44:50.910 -04:00 [Fatal] Failed to start application.
System.ServiceModel.FaultException`1[System.ServiceModel.ExceptionDetail]: Cannot access a disposed object.
Object name: 'System.ServiceModel.Channels.ServiceChannel'. (Fault Detail is equal to An ExceptionDetail, likely created by IncludeExceptionDetailInFaults=true, whose value is:
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'System.ServiceModel.Channels.ServiceChannel'.

Server stack trace: 
   at System.ServiceModel.Channels.CommunicationObject.ThrowIfDisposedOrNotOpen()
   at System.ServiceModel.Channels.ServiceChannel.Call(String action, Boolean oneway, ProxyOperationRuntime operation, Object[] ins, Object[] outs, TimeSpan timeout)
   at System.ServiceModel.Channels.ServiceChannelProxy.InvokeService(IMethodCallMessage methodCall, ProxyOperationRuntime operation)
   at System.ServiceModel.Channels.ServiceChannelProxy.Invoke(IMessage message)

Exception rethrown at [0]: 
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
   at ChocolateyGui.IIpcServiceCallbacks.LogMessage(StreamingLogMessage message)
  ...).

```

**Comment by RichiCoder1** [comment]
I tentatively believe this will be solved by #466.

**Comment by TheVakman** [comment]
Is #466 included in 0.14.0-unstable0343?

If so, this is not resolved for me.

**Comment by gep13** [comment]
@TheVakman to answer your question...

Anything that is merged into the develop branch will produce a build/release on our MyGet feed.  So pulling the latest version from MyGet, which is currently `0.14.0-unstable0343` should include all code on the develop branch.

**Comment by gep13** [comment]
@TheVakman I have just tried to run ChocolateyGUI as a non-admin and I am not seeing the error that you are describing.

Can you please provide some more information about your current setup, i.e:

* Operating System
* Chocolatey Version
* Chocolatey Extension Version
* License Level
* etc

Hopefully that will help figure out what the issue is.

**Comment by TheVakman** [comment]
@gep13 I seem to have a different error upon launching as a non-admin now as well. I'll attach it below.

- OS: Windows 10 v1607
- Chocolatey Version: 0.10.8
- Chocolatey Core Extensions: 1.3.1
- Chocolatey Licensed Extension: 1.12.3
- Chocolatey Agent: 0.7.0

![newissue](https://user-images.githubusercontent.com/18545156/30862890-6ecfc01e-a29d-11e7-9ba8-4d23c9d7ae27.PNG)

**Comment by RichiCoder1** [comment]
@TheVakman Is the being performed in a management or enterprise environment? Is there an AV or policy that's blocking user programs from executing other processes?

**Comment by TheVakman** [comment]
That's a good point. I have ChocolateyGUI and the chocolatey exe allowed but I'll check as it could be due to the different path that's used now. I'll report back shortly, thanks.

**Comment by TheVakman** [comment]
I've removed our security software for testing. The issue still persists.

Any other suggestions?

**Comment by gep13** [comment]
@TheVakman we know that there is still at least one issue with ChocolateyGUI locating the license file when running as a non-administrator user, and this issue is currently being worked on.  It could be that this is related to this issue that you are seeing.  Let us get to a point where we know that this part is working, and we can circle back around to verify this issue.

**Comment by gep13** [comment]
@TheVakman can you please install 1.12.4 of the chocolatey.extension (https://chocolatey.org/docs/release-notes-licensed#september-27-2017) and install the latest Chocolatey GUI from the MyGet feed then try this again?

We believe that we have solved the underlying problems.  If you can report back and let us know if you are still having problems.

For now, I am going to go ahead and close this off, but feel free to re-open if still a problem.

**Comment by TheVakman** [comment]
I am unable to re-open the issue (I assume permissions on this GitHub repo) but it is definitely still there on 0345 and 0351 of ChocolateyGUI.

However, I updated my extension and ChocolateyGUI 0.14.0345 was working with Self Service.
As per your suggestion, I installed the latest. This means an uninstall of 0345 after as upgrading seems to end up with two versions (I'll log a bug for this) but now I have the same crash as before.

So I uninstalled, deleted any remaining files for a fresh installation, reinstalled the latest, crashed. I opened as an Administrator, checked the box for Self Service again (since I had uninstalled completely/doesn't seem to honour chocolatey setting), saved, relaunched as a non-admin and the same crash occurs.

**Comment by RichiCoder1** [comment]
Any ideas @ferventcoder ?

**Comment by gep13** [comment]
@TheVakman would you be able to provide the full log from the `c:\programdata\chocolateygui\logs` folder?

**Comment by gep13** [comment]
> @TheVakman said....
I am unable to re-open the issue (I assume permissions on this GitHub repo)

That is very strange.  As the author of the issue, I thought you were able to re-open it.  Looks like @RichiCoder1 took care of that for you though.

**Comment by ferventcoder** [comment]
I've been able to get this one to work. `choco upgrade chocolateygui -y --pre --version 0.14.0-unstable0345 -s https://www.myget.org/F/chocolateygui/ --allow-downgrade` - but you must uninstall the current GUI first so you don't accidentally open the old one thinking it is the new one.

**Comment by TheVakman** [comment]
That's what I ran on the machines to reinstall 0345 after I upgraded and it ceased to be working. Although I didn't have "--allow-downgrade" because I had uninstalled the other. I tried it with that now though and I'm still unable to replicate the initial success.

Is it an order of operations perhaps? I already had 0345 installed and then I upgraded the extension when I had it working.

**Comment by TheVakman** [comment]
@ferventcoder has confirmed that it was only occurring over RDP. This will be fixed in 1.12.6 of the extension, by the looks of it.

**Comment by ferventcoder** [comment]
### More information

* When you are using Chocolatey commercial editions over RDP, it will run `change user /install|/execute` for you automatically.
* Chocolatey GUI is 32-bit based and C:\Windows\System32 hit filesystem redirection to C:\Windows\SysWOW64.
* SysWOW64 doesn't contain change.exe.

Fixes for Chocolatey Licensed Extension v1.12.6:
* Search in the c:\windows\SysNative virtual directory for change.exe to run it.
* Ensure that if the terminal services switch fails, it doesn't break Chocolatey.

**Comment by gep13** [comment]
@TheVakman very glad to hear that you have got to the bottom of this.
